File: src/main/java/VetCare/Back/model/entities/Prontuario.java
package VetCare.Back.model.entities;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "prontuarios")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Prontuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pet_id", nullable = false)
    private Pet pet;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "veterinario_id", nullable = false)
    private Veterinario veterinario;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "agendamento_id")
    private Agendamento agendamento;

    private LocalDateTime dataAtendimento;

    @Column(columnDefinition = "TEXT")
    private String descricao;
    @Column(columnDefinition = "TEXT")
    private String diagnostico;
    @Column(columnDefinition = "TEXT")
    private String tratamento;
    @Column(columnDefinition = "TEXT")
    private String prescricao;
}

File: front/app/types/prontuario.ts
export interface ProntuarioRequest {
  dataAtendimento: string;
  descricao?: string;
  diagnostico?: string;
  tratamento?: string;
  prescricao?: string;
  pet: { id: number };
  agendamento?: { id: number };
}

export interface ProntuarioResponse {
  id: number;
  petId: number;
  veterinarioId: number;
  agendamentoId: number | null;
  dataAtendimento: string;
  descricao: string;
  diagnostico: string;
  tratamento: string;
  prescricao: string;
}
